server/coordinator/scheduler/scheduler_manager.go

```diff
@@ -167,11 +174,17 @@ func (callback *schedulerWatchCallback) OnShardExpired(ctx context.Context, even
 
 // Schedulers should to be initialized and registered here.
 func (m *ManagerImpl) initRegister() {
-	assignShardScheduler := NewAssignShardScheduler(m.factory, m.nodePicker)
-	m.registerScheduler(assignShardScheduler)
+	if m.topologyType == TopologyTypeDynamic {
```

Comment:
A better way to collect the different schedulers for different topology types, and then register them one by one to avoid calling `m.registerScheduler` again and again.